ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+    m_deviceFeatureLayer->clearSelection();
+    m_lineFeatureLayer->clearSelection();
+
+    m_busy = true;
+    emit isBusy();
+
+    m_message = QString("Identifying feature to edit...");
+    emit messageChanged();
+
+    if (m_map->loadStatus() != LoadStatus::Loaded)
+      return;
+
+    m_clickPoint = m_mapView->screenToLocation(mouseEvent.position().x(), mouseEvent.position().y());
+
+    // Perform an identify to determine if a user tapped on a feature.
+    m_mapView->identifyLayersAsync(mouseEvent.position(), 5.0, false).then(this, [this](const QList<IdentifyLayerResult*>& results)
+    { if(!results.isEmpty())
+      {
+        for(IdentifyLayerResult* result : results)
+        {
+          if(result && (result->layerContent()->name() == m_deviceTableName || result->layerContent()->name() == m_lineTableName))
+          {
+            m_feature = static_cast<ArcGISFeature*>(qAsConst(result)->geoElements()[0]);
+            break;
+          }
+        }
+      }
+      else
+      {
+        m_message = QString("No feature identified. Tap on a feature to edit.");
+        emit messageChanged();
+
+        m_busy = false;
+        emit isBusy();
+
+        return;
+      }
+      m_updateFieldName = m_feature->featureTable()->tableName() == m_deviceTableName ? m_deviceStatusField : m_nominalVoltageField;
+      m_field = m_feature->featureTable()->field(m_updateFieldName);
+      m_codedValueDomain = static_cast<CodedValueDomain>(m_field.domain());
+      m_codedValues = m_codedValueDomain.codedValues();
+      if (m_field.isEmpty() || m_codedValues.empty())
+      {
+        return;
+      }
+
+      if (m_feature->loadStatus() != LoadStatus::Loaded)
+        m_feature->load();
+
+      m_featureToEdit = m_feature;
+
+      // Select the feature.
+      if (m_updateFieldName == m_nominalVoltageField)
+        m_lineFeatureLayer->selectFeature(m_featureToEdit);
+      else if (m_updateFieldName == m_deviceStatusField)
+        m_deviceFeatureLayer->selectFeature(m_featureToEdit);
+
+      if(!m_choices.isEmpty())
+        m_choices.clear();
+
+      m_choices.append("");
+      for (const CodedValue& codedValue : m_codedValues)
+      {
+        if(codedValue.code() == m_featureToEdit->attributes()->attributeValue(m_updateFieldName))
+
+        m_choices.replace(0, codedValue.name());
+        else
+            m_choices.append(codedValue.name());
+      }
+      emit attributeValue();
+      emit addChoices();
+
+      m_message = QString("Select a new ") + m_updateFieldName;
+
+      m_clearBtn = true;
+      emit clearBtn();
+
+      m_isUpdateVisible = true;
+      emit isUpdateVisible();
+
+      m_busy = false;
+      emit isBusy();
+
+      emit messageChanged();
+      emit updateFieldName();
+    });
+  });
+}
+
+void ValidateUtilityNetworkTopology::onApplyEdits(QString choice)
+{
+  m_serviceGeodatabase = dynamic_cast<ServiceFeatureTable*>(m_featureToEdit->featureTable())->serviceGeodatabase();
```

Comment:
```suggestion
  m_serviceGeodatabase = static_cast<ServiceFeatureTable*>(m_featureToEdit->featureTable())->serviceGeodatabase();
```
If you are certain the cast will work, you `static_cast`

Reply:
done